elixir/apps/domain/lib/domain/accounts/account/changeset.ex

```diff
@@ -2,22 +2,61 @@ defmodule Domain.Accounts.Account.Changeset do
   use Domain, :changeset
   alias Domain.Accounts.Account
 
-  def create_changeset(attrs) do
-    %Account{}
+  def changeset(account, attrs) do
+    account
     |> cast(attrs, [:name, :slug])
     |> validate_required([:name])
-    |> validate_length(:name, min: 1, max: 255)
+    |> validate_name()
+    |> trim_name_whitespace()
+    |> set_slug_default()
+    |> downcase_slug()
+    |> remove_spaces_from_slug()
```

Comment:
I think it's better if we validate and don't accept invalid slug values with `validate_format/3` rather than trying to clean it up. In this case, once we start allowing people to pick their own slugs - we will be able to give users feedback instead of just filtering and saving (that can be ambiguous).

Reply:
Makes sense.  I wrote this when we were thinking we'd just take the `name` and create a `slug` out of it, but now that we're generating a random slug I agree with what you've described.

Reply:
This has been removed.